app/scheduler/onlinetesting/EvaluatePhase1ResultJob.scala

```diff
@@ -30,14 +36,41 @@ object EvaluatePhase1ResultJob extends EvaluatePhase1ResultJob {
 trait EvaluatePhase1ResultJob extends SingleInstanceScheduledJob with EvaluatePhase1ResultJobConfig {
   val evaluateService: EvaluatePhase1ResultService
 
+  override implicit val ec = ExecutionContext.fromExecutor(
+    new ThreadPoolExecutor(2, 2, 180, TimeUnit.SECONDS, new ArrayBlockingQueue(batchSizeLimit))
+  )
+
   def tryExecute()(implicit ec: ExecutionContext): Future[Unit] = {
-    evaluateService.nextCandidateReadyForEvaluation.map {
-      case Some((app, passmarkSettings)) =>
-        Logger.debug(s"Phase1 evaluation for applicationId=${app.applicationId} against passmarkVersion=${passmarkSettings.version}")
-        evaluateService.evaluate(app, passmarkSettings)
+    evaluateService.nextCandidatesReadyForEvaluation(batchSizeLimit) flatMap {
+      case Some((apps, passmarkSettings)) =>
+        evaluateInBatch(apps, passmarkSettings)
       case None =>
-        Logger.info("Passmark settings or Application to evaluate phase1 result not found")
+        Logger.info("Passmark settings or an application to evaluate phase1 result not found")
+        Future.successful(())
+    }
+  }
+
+  private def evaluateInBatch(apps: List[ApplicationPhase1ReadyForEvaluation], passmarkSettings: Phase1PassMarkSettings): Future[Unit] = {
+    Logger.debug(s"Evaluate Phase1 Job found ${apps.size} application(s), the passmarkVersion=${passmarkSettings.version}")
+    val evaluationResultsFut = FutureEx.traverseToTry(apps) { app =>
```

Comment:
Shall we just use Future.traverse instead of traverseToTry, as we are wrapping with Try in the function already